spring-integration-rsocket/src/main/java/org/springframework/integration/rsocket/RSocketConnectedEvent.java

```diff
@@ -42,10 +44,10 @@ public class RSocketConnectedEvent extends IntegrationEvent {
 
 	private final DataBuffer data;
 
-	private final RSocketRequester requester;
+	private final @Nullable RSocketRequester requester;
 
 	public RSocketConnectedEvent(Object source, Map<String, Object> headers, DataBuffer data,
-			RSocketRequester requester) {
+			@Nullable RSocketRequester requester) {
```

Comment:
I don't think this is correct.
I think we should fix `ServerRSocketMessageHandler.handleConnectionSetup()` to assert that `rsocketRequester` is not null from those headers.